src/app/shared/users.service.ts

```diff
@@ -1,33 +1,58 @@
 import { Injectable } from '@angular/core';
-import { usersList } from './users.mock';
 import { User } from './users.model';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UsersService {
 
-  constructor() { }
+  mainURL: string = 'http://localhost:8080/';
+  usersMainURL: string = `${this.mainURL}users/`;
 
-  getAllUsers(): User[] {
-    return usersList;
+  constructor(private http: HttpClient) { }
+
+  getAllUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL);
+  }
+
+  getActiveUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL).pipe(
+      map(usersList => usersList.filter(user => !user.isDeleted))
+    );
   }
 
-  getActiveUsers(): User[] {
-    return usersList.filter(user => user.isDeleted === false);
+  getDeletedUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL).pipe(
+      map(usersList => usersList.filter(user => user.isDeleted))
+    );
   }
 
-  getDeletedUsers(): User[] {
-    return usersList.filter(user => user.isDeleted === true);
+  getUserDetails(userId: string): Observable<User> {
+    return this.http.get<User>(this.usersMainURL + userId);
   }
 
-  getUserDetails(userId: number): User {
-    return usersList.find(user => user.id === userId);
+  changeStatus(userData: User): Observable<User> {
+    return this.http.put<User>(this.usersMainURL + userData.id, {
+      isDeleted: !userData.isDeleted
+    });
   }
 
-  changeStatus(userId: number): void {
-    let index = usersList.findIndex(user => user.id === userId);
-    usersList[index].isDeleted = !usersList[index].isDeleted;
+  createUser(userData: User): Observable<User> {
+    return this.http.post<User>(this.usersMainURL, userData);
   }
 
+  updateUser(userData: Partial<User>) {
+    console.log("from updateservice", userData);
```

Comment:
Remove `console.log` and commented code from the files before you commit the code

Reply:
remove the console.log